src/bentoml/_internal/monitoring/otlp.py

```diff
@@ -0,0 +1,220 @@
+                "monitor_name": "iris_classifier_prediction",
+                "schema": {
+                    "sepal length": {
+                        "name": "sepal length",
+                        "role": "feature",
+                        "type": "numerical",
+                    },
+                    "sepal width": {
+                        "name": "sepal width",
+                        "role": "feature",
+                        "type": "numerical",
+                    },
+                    "petal length": {
+                        "name": "petal length",
+                        "role": "feature",
+                        "type": "numerical",
+                    },
+                    "petal width": {
+                        "name": "petal width",
+                        "role": "feature",
+                        "type": "numerical",
+                    },
+                    "pred": {"name": "pred", "role": "prediction", "type": "categorical"},
+                },
+            },
+        }
+
+    """
+
+    PRESERVED_COLUMNS = (COLUMN_TIME, COLUMN_RID, COLUMN_META) = (
+        "timestamp",
+        "request_id",
+        "bento_meta",
+    )
+
+    def __init__(
+        self,
+        name: str,
+        endpoint: str | None = None,
+        insecure: bool | str | None = None,
+        credentials: str | None = None,
+        headers: str | None = None,
+        timeout: int | str | None = None,
+        compression: str | None = None,
+        meta_sample_rate: float = 1.0,
+        **_: t.Any,
+    ) -> None:
+        """
+        Initialize the monitor.
+
+        Args:
+            name: The name of the monitor.
+            endpoint: The endpoint to send the data to.
+            insecure: Whether to use insecure connection.
+            credentials: The credentials to use.
+            headers: The headers to use.
+            timeout: The timeout to use.
+            compression: The compression to use.
+        """
+        super().__init__(name, **_)
+
+        self.endpoint = endpoint
+        self.insecure = insecure
+        self.credentials = credentials
+        self.headers = headers
+        self.timeout = timeout
+        self.compression = compression
+        self.meta_sample_rate = meta_sample_rate
+
+        self.logger_provider: LoggerProvider | None = None
+        self.data_logger: logging.Logger | None = None
+        self._schema: dict[str, dict[str, str]] = {}
+        self._will_export_schema = False
+
+    def _init_logger(self) -> None:
+        self.logger_provider = LoggerProvider(
+            resource=Resource.create(
+                {
+                    "service.name": f"{component_context.bento_name}:{self.name}",
+                    "service.instance.id": "{component_context.bento_version}",
+                }
+            ),
+        )
+        set_logger_provider(self.logger_provider)
+
+        if self.endpoint is not None:
```

Comment:
We shouldn't override user env if they set themselves right?

Reply:
It has the same behavior as open telemetry implementation now.

Reply:
This isn't the behavior now - I don't know if it was at the time: https://github.com/open-telemetry/opentelemetry-python/blob/ea15f72f6678ff23c86511c095094cb3e4bafc44/exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/_log_exporter/__init__.py#L75-L94